src/utils/execution-order.ts

```diff
@@ -0,0 +1,147 @@
+import ExternalModule from '../ExternalModule';
+import Module from '../Module';
+import { randomUint8Array, Uint8ArrayEqual, Uint8ArrayXor } from './entryHashing';
+import error from './error';
+import { relative } from './path';
+import relativeId from './relativeId';
+
+interface OrderedExecutionUnit {
+	execIndex: number;
+}
+
+const compareExecIndex = <T extends OrderedExecutionUnit>(unitA: T, unitB: T) =>
+	unitA.execIndex > unitB.execIndex ? 1 : -1;
+
+export function sortByExecutionOrder(units: OrderedExecutionUnit[]) {
+	units.sort(compareExecIndex);
+}
+
+export function analyzeModuleExecution(
+	entryModules: Module[],
+	generateChunkHashes: boolean,
```

Comment:
Note that these hashes are not the same as "chunk hashes" as chunk hashes should be content-based.

So we have two concept of chunk hash:
1. The content-based hash of the inner chunk content with output options
2. The graph colouring hash

I prefer to use `chunkHash` to refer to (1) as it is what most users would naively expect I believe.

So either `colouringHash` or `graphHash` or similar for this would be a more useful name I think.

Reply:
Good point!